### PR TITLE
Add opt-in support for chimeric ProForma parsing

### PR DESCRIPTION
**Pull request by josiehong**
Adds opt-in parsing support for chimeric ProForma strings using `chimeric=True` related to #209. 

This keeps the default single-sequence behavior unchanged while allowing top-level `+` to separate chimeric peptidoform components when explicitly requested.

## Changes

- Add `chimeric` option to `Parser`, `proforma.parse`, and `ProForma.parse`.
- Return a list of parse results / `ProForma` objects when `chimeric=True`.
- Share fixed modifications, isotope labels, and peptidoform names across chimeric components.
- Preserve the current default behavior for `+`, with an error message suggesting `chimeric=True`.
- Add tests for opt-in parsing, shared metadata, empty components, charge/adduct boundaries, and shared fixed modifications.
- Document chimeric parsing and component-wise use with existing downstream APIs.

### CODE REVIEW

**Comment by mobiusklein** [comment]
@levitsky can you please approve workflow runs for this PR? I don't have permission.

File: tests/test_proforma.py [hunk 1]
```diff
@@ -138,6 +138,60 @@ def test_charge_adducts(self):
...
+        self.assertEqual(len(parsed[1][1]['fixed_modifications']), 1)
+
+        forms = ProForma.parse(seq, chimeric=True)
+        aa_comp = std_aa_comp.copy()
+        aa_comp['cam'] = Composition(formula='H3C2NO')
+        self.assertEqual(forms[0].composition(), Composition(sequence='AcamC', aa_comp=aa_comp))
+        self.assertEqual(forms[1].composition(), Composition(sequence='camCcamC', aa_comp=aa_comp))
+
+    def test_chimeric_shared_names_and_isotopes(self):
+        parsed = parse('(>sample)<13C>AC+CC', chimeric=True)
```
Comment (mobiusklein) [inline tests/test_proforma.py:183]: This bit is tricky. Names defined with `>` and `>>` are specific to a single peptidoform and peptidoform ion respectively. Only `>>>`  names apply to all peptidoforms in the set. This is section 8.2 of the forthcoming specification: https://docs.google.com/document/d/1HaSOMNmdFCxy0G6lq1Yci03z-4esmQoe_hSyoqTeRnc/edit?tab=t.0

So `AC` would have the name `sample`, but `CC` would have no name.

The example from the specification is `(>Trypsin)AANSIPYQVSLNS+(>Keratin)AKEQFERQTA`

**Review by mobiusklein — APPROVED** [review]
This looks great other than the name sharing thing (which is trivial and can be fixed at a later date).

I'm happy to bundle that fix in with other changes when I do a coverage overhaul.
